fenix/docs/architectureexample/HistoryStore.kt

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This is example code for the 'Simplified Example' section of
+// /docs/architecture-overview.md
+class HistoryStore(
+    private val initialState: HistoryState,
+    private val middleware: List<Middleware<HistoryState, HistoryAction>>
+) : Store<HistoryState, Reducer<HistoryState, HistoryAction>>(initialState, middleware, ::reducer) {}
+
+sealed class HistoryAction {
+    object Init : HistoryAction()
+    data class ItemsChanged(val items: List<History>) : HistoryAction()
+    data class DeleteItems(val items: List<History>) : HistoryAction()
+    data class DeleteFinished() : HistoryAction()
+    data class ToggleItemSelection(val item: History) : HistoryAction()
+    data class OpenItem(val item: History) : HistoryAction()
+}
+
+data class HistoryState(
+    val items: List<History>,
+    val selectedItems: List<History>,
+    val itemsBeingDeleted: List<History>,
+    companion object {
+        val initial = HistoryState(
+            items = listOf(),
+            selectedItems = listOf(),
+            itemsBeingDeleted = listOf(),
+        )
+    }
+)
```

Comment:
Can create a derived property of this state, so this logic is removed from the UI and when the state is updated, the displayItems are present as it's based on two other existing state values. 

```
    val displayItems = items.filter { item ->
        item !in itemsBeingDeleted
    }
```

Reply:
Nice 😎